connector/session.py

```diff
@@ -202,6 +202,11 @@ def rollback(self):
 
     def close(self):
         """ Close the cursor """
+        try:
+            # Try to recompute fields before closing the cursor
+            self.env['ir.model'].recompute()
+        except:
+            self.rollback()
```

Comment:
I am not comfortable with this change, because I had great trouble to understand why it happens and how this fix resolves it. `retry_postpone` opens a new transaction and a new `Environment` (https://github.com/OCA/connector/blob/9.0/connector/session.py#L73), so I assumed that the `todo` added before the `OperationalError` would not leak into the new transaction. Happens I was wrong, because `todo` is shared by all environments (https://github.com/odoo/odoo/blob/9.0/openerp/api.py#L852) which is rather unexpected. 

However, this fix seems a bit "radical" to me, shouldn't we prevent this to happen at the first place and clear the Environment (`Environment.clear()` https://github.com/odoo/odoo/blob/9.0/openerp/api.py#L821-L826) only at the moment when we catch the `OperationalError`?